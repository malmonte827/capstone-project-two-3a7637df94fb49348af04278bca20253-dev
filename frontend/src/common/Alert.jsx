import React from "react";

/** Alert component provides alert feedback messages to other components
 *
 * { SignupForm , LoginForm  }
 */

function Alert({ type = "danger", messages = [] }) {
    return (
        <div class={`alert alert-${type}`} role="alert">
            {messages.map((error) => (
                <p className="mb-0 small" key={error}>
                    {error}
                </p>
            ))}
        </div>
    );
}

export default Alert;
