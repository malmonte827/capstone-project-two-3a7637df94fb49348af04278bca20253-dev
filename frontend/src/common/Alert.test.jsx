import React from "react";
import Alert from "./Alert";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

describe("<Alert />", () => {
    it("renders without crashing", () => {
        render(<Alert />);
    });

    it("matches snapshot", () => {
        const { asFragment } = render(<Alert />);
        expect(asFragment()).toMatchSnapshot();
    });

    describe("Displays messages correctly", () => {
        it("displays the correct message when messages array is passed", () => {
            const messages = ["Error occurred!", "Please try again."];
            render(<Alert messages={messages} />);

            messages.forEach((msg) => {
                expect(screen.getByText(msg)).toBeInTheDocument();
            });
        });
    });
});
