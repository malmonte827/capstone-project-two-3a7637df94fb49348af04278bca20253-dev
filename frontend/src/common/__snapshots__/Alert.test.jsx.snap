// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Alert /> > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  />
</DocumentFragment>
`;
